internal-enrichment/shodan/src/shodanImport.py

```diff
@@ -29,310 +40,466 @@ def __init__(self):
+        )
+
+    def _get_entity_in_opencti(self, data):
+        opencti_entity = self.helper.api.stix_cyber_observable.read(
+            id=data["entity_id"]
+        )
+        if opencti_entity is None:
+            raise ValueError(
+                "Observable not found (or the connector does not has access to this observable, check the group of the connector user)"
+            )
+        return opencti_entity
 
-        # Generate Domain Desc Block
-        Domains = "Domains:"
-        for domain in shodanHostResponse["domains"]:
-            Domains = Domains + f"\n  - {domain}"
+    def _extract_and_check_markings(self, entity):
+        tlp = "TLP:CLEAR"
+        for marking_definition in entity["objectMarking"]:
+            if marking_definition["definition_type"] == "TLP":
+                tlp = marking_definition["definition"]
+
+        is_valid_max_tlp = OpenCTIConnectorHelper.check_max_tlp(tlp, self.max_tlp)
+        if not is_valid_max_tlp:
+            raise ValueError(
+                "Do not send any data, TLP of the observable is greater than MAX TLP"
+            )
+        return tlp
+
+    def _generate_stix_relationship(
+        self, source_ref, stix_core_relationship_type, target_ref
+    ):
+        return stix2.Relationship(
+            id=StixCoreRelationship.generate_id(
+                stix_core_relationship_type, source_ref, target_ref
+            ),
+            relationship_type=stix_core_relationship_type,
+            source_ref=source_ref,
+            target_ref=target_ref,
+            created_by_ref=self.shodan_identity["standard_id"],
+            confidence=self.helper.connect_confidence_level,
+        )
 
+    @staticmethod
+    def _generate_description(data):
         # Generate Services Desc Block
-        Services = "Services:\n"
-        for service in shodanHostResponse["data"]:
-            serviceData = service["data"].strip()
-            Services = (
-                Services + f'\n**{str(service["port"])}:**\n```\n{serviceData}\n```'
+        services_desc = "Services:\n"
+        for service in data["data"]:
+            service_data = service["data"].strip()
+            services_desc = (
+                services_desc
+                + f'\n**{str(service["port"])}:**\n```\n{service_data}\n```'
             )
 
             if "opts" in service:
                 if "heartbleed" in service["opts"]:
-                    Services = (
-                        Services + f'\nHEARTBLEED: {service["opts"]["heartbleed"]}'
+                    services_desc = (
+                        services_desc + f'\nHEARTBLEED: {service["opts"]["heartbleed"]}'
                     )
-            Services = Services + "\n------------------"
+            services_desc = services_desc + "\n------------------"
+
+        global_description = f"""
+    **ISP:** {data["isp"]}
+    
+    **OS:** {str(data["os"]) if data["os"] is not None else "-"}
+    
+    --------------------------
+    {services_desc}
```

Comment:
The indentation creates wrong display:

![image](https://github.com/OpenCTI-Platform/connectors/assets/75518128/e7abb304-a912-409a-8334-2f2470458703)

```suggestion
**ISP:** {data["isp"]}
    
**OS:** {str(data["os"]) if data["os"] is not None else "-"}
    
--------------------------
{services_desc}
```